src/workloads/execution/FilterWithLogicalExpressions.yml

```diff
@@ -0,0 +1,162 @@
+          a: 1
+          b: 1
+          c: 1
+          d: 1
+
+# Phase 1: Ensure all data is synced to disk.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Phase 2: Run query with wide expression.
+- Name: RunWideQuery
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 10
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {
+              $expr: {
+                $or: [
+                  {$eq: ["$a", 10]},
+                  {$eq: ["$b", 11]},
+                  {$eq: ["$c", 12]},
+                  {$eq: ["$d", 13]},
+                  {$eq: ["$a", 14]},
+                  {$eq: ["$b", 15]},
+                  {$eq: ["$c", 16]},
+                  {$eq: ["$d", 17]},
+                  {$eq: ["$a", 18]},
+                  {$eq: ["$b", 19]},
+                ]
+              }
+            }
+
+# Phase 3: Run query with deep expression.
+- Name: RunDeepQuery
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 10
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {
+              $expr: {
+                $or: [
+                  {
+                    $and: [
+                      {
+                        $or: [
+                          {$eq: ["$a", 10]},
+                          {$eq: ["$b", 11]}
+                        ]
+                      },
+                      {
+                        $or: [
+                          {$eq: ["$c", 12]},
+                          {$eq: ["$d", 13]}
+                        ]
+                      },
+                    ]
+                  },
+                  {
+                    $and: [
+                      {
+                        $or: [
+                          {$eq: ["$a", 14]},
+                          {$eq: ["$b", 15]}
+                        ]
+                      },
+                      {
+                        $or: [
+                          {$eq: ["$c", 16]},
+                          {$eq: ["$d", 17]}
+                        ]
+                      },
+                    ]
+                  }
+                ]
+              }
+            }
+
+# Phase 4: Run query with expression which goes both wide and deep.
+- Name: RunMixedQuery
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [4]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 10
```

Comment:
Similar to @allenwux comment earlier 1 thread and 10 repetitions seems small. Did you schedule a number of patch-builds with identical configuration and git revision, then analyze the distribution of results? If so, did you see low result variability?

Reply:
Number 10 was chosen to speedup the local debugging. This is the first time I hear about the requirement to perform distribution analysis through a number of Evergreen runs. It makes sense, but I would prefer to do this tuning when we settle on the types of queries this workload should include.